library/src/hcc_detail/hipsparse.cpp

```diff
@@ -13370,6 +13370,36 @@ hipsparseStatus_t hipsparseSpMV_bufferSize(hipsparseHandle_t           handle,
                                            hipsparseSpMVAlg_t          alg,
                                            size_t*                     bufferSize)
 {
+    if(handle == nullptr)
+    {
+        return rocSPARSEStatusToHIPStatus(rocsparse_status_invalid_handle);
+    }
+
+    if(matA == nullptr || vecX == nullptr || vecY == nullptr || alpha == nullptr || beta == nullptr
+       || bufferSize == nullptr)
+    {
+        return rocSPARSEStatusToHIPStatus(rocsparse_status_invalid_pointer);
+    }
+
+    std::cout << "hipsparseSpMV_bufferSize called" << std::endl;
+
+    *bufferSize = 4;
```

Comment:
0?

Reply:
Currently cannot be zero because that would mean doing:

void* temp_buffer = nullptr;
hipMalloc(&temp_buffer, buffer_size);

would still result in temp_buffer being nullptr and in the rocsparse code temp_buffer = nullptr means (when passed to rocsparse_spmv() ) calculate the buffer size. So we need to do our refactor to spmv in rocsparse at the next major release to add the 'stage' enum before we can set the buffer size to 0.

Reply:
Thanks ! again ill-specified spmv ....